types/src/transaction/mod.rs

```diff
@@ -592,16 +592,54 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum GovernanceRole {
+    LibraRoot,
+    TreasuryCompliance,
+    Validator,
+    ValidatorOperator,
+    DesignatedDealer,
+    NonGovernanceRole,
+}
+
+impl GovernanceRole {
+    pub fn from_role_id(role_id: u64) -> Self {
+        use GovernanceRole::*;
+        match role_id {
+            0 => LibraRoot,
+            1 => TreasuryCompliance,
+            2 => DesignatedDealer,
+            3 => Validator,
+            4 => ValidatorOperator,
+            _ => NonGovernanceRole,
+        }
+    }
+
+    pub fn priority(&self) -> u64 {
```

Comment:
nit: maybe a comment that the larger the priority value, the more greatly it is prioritized?